sdk/src/main/scala/com/horizen/forge/Forger.scala

```diff
@@ -30,19 +30,37 @@ class Forger(settings: SidechainSettings,
   val timeoutDuration: FiniteDuration = settings.scorexSettings.restApi.timeout
   implicit val timeout: Timeout = Timeout(timeoutDuration)
 
-  @volatile private var forgingIsActive: Boolean = false
 
   val consensusMillisecondsInSlot: Int = params.consensusSecondsInSlot * 1000
-  val forgingInitiator: Runnable = () => {
-    while (true) {
-      if (forgingIsActive) {
-        tryToCreateBlockNow()
+  def forgingInitiatorTimerTask: TimerTask = new TimerTask {override def run(): Unit = tryToCreateBlockNow()}
+  var timerOpt: Option[Timer] = None
+
+  def startTimer(): Unit = {
+    this.synchronized{
```

Comment:
`this.synchronized{` can be removed. Timer access is controlled by Akka receivable message queue.